### DISCUSSION TITLE
MacOS issue

### DISCUSSION THREAD

**Discussion by raphaelquast** [raphaelquast/beatstep#80] [General]
A general discussion about the MacOS issue...

**Comment by li-il-li** [comment]
Hi I found some time yesterday to check the messages transmitted over USB from my Mac using a Raspberry Pi.
Here is the [repo](https://github.com/dwi-iwb/RPi-MIDI-Intercepter) with the script and the log.
Happy to run many other test.

**Reply by raphaelquast** [reply]
@li-il-li hey, just noticed that I somehow missed your response here... 
(maybe the notification was identified as spam or so...) sorry!
The repo is not accessible any more so I can't have a look.

**Comment by pixel-123** [comment]
I come from the previous longer thread about this issue (https://github.com/raphaelquast/beatstep/issues/58) and I am still hopeful this will be resolved some day! Also seeing if new versions of Live 11 make it work. So far no. 

BTW, is there any combination of Live versions and MacOS versiosn that have been shown to work well? I probably wouldn't mind trying that combination if it works...

**Reply by raphaelquast** [reply]
@Helder-boop 
Hey, sorry for the late response... and I'm really sorry but I have to disappoint you again...
At the moment there has not been any progress in with respect to the MacOS issue.
I don't have a lot of time investigating the problems right now and I still don't own a Mac so I can't reproduce the issue (and consequently I don't know what to change to make it go away)

If you can try some combinations of Live11 and the script it would be nice to see, but at the moment I can't even pinpoint if the problem is somewhat related to the Live version.

**Reply by pixel-123** [reply]
Ok, no worries :) Thank you for the reply. I'll keep hopeful someday it will work :) someone will find a solution :)
Best wishes!

**Reply by raphaelquast** [reply]
I'll keep my eyes open 😅 if anything relevant pops up I'll definitely give it another try!

**Comment by pixel-123** [comment]
Thanks! I believe! Hehehe
Cheers.

> I'll keep my eyes open 😅 if anything relevant pops up I'll definitely give it another try!

**Comment by subduct** [comment]
@raphaelquast I've recently got myself a beatstep and I wondered if I can help diagnosing/fixing?
I have a Mac and Ableton 11/12 (Mac is however older and only running big sur 11.4)
I do not have any programming experience whatsoever!
Also appreciate this is a couple of years and you may have other priorities/no time for this.

**Comment by raphaelquast** [comment]
@subduct 
Hey! Yep... my last encounter with this issue is quite some time ago...
We were never able to figure out completely where things go wrong on MacOs... somehow MIDI messages got lost on the way but there was no indication anywhere why this happened... 

It might have been an error that has resolved itself until now... so maybe just give it a try if it works? 
(I still don't own a Mac so I simply can't tell)

Without any programming experience its a bit difficult to help out here since I don't have the time to do the coding... 
(But that does not mean that you can't give it a try! I also had no experience with Ableton MIDI remote scripts when I started developing this)

I'm happy to help out wherever I can but my time is quite limited currently... (and I'd definitely merge any PullRequest that solves this)

**Reply by subduct** [reply]
Thank you! 
I'll try things out and see where I get to.
